pctiler/pctiler/reader.py

```diff
@@ -49,6 +67,10 @@ class MosaicSTACReader(pgstac_mosaic.CustomSTACReader):
 
     reader: Type[BaseReader] = attr.ib(default=CustomCOGReader)
 
+    # We make request an optional attribute to avoid re-writing
+    # the whole list of attribute
+    request: Optional[Request] = attr.ib(default=None)
```

Comment:
With `request` now optionally on `MosaicSTACReader`, I expected to be able to access `self.reader.request` within methods of `PGSTACBackend` (like [`assets_for_tiles`](https://github.com/microsoft/planetary-computer-apis/blob/main/pctiler/pctiler/reader.py#L80-L85)), but I get the error 

` AttributeError: type object 'MosaicSTACReader' has no attribute 'request'`

The type annotations suggest this is possible, is that the expected usage? I can confirm that `ReaderParams` was initialized with a request correctly.

Reply:
can you share some code, I have hard to to follow 😓

Reply:
Sure, sorry! One use case for having the request available is to use it for our logging within backend/reader methods, e.g., grabbing a trace id from the header so all logs can be associated with a single request. Usage would be like this: https://github.com/microsoft/planetary-computer-apis/commit/f46100939cb44c6440ec07824ca950e2b35da653, but it doesn't seem like request is available from there.

Reply:
latest commit should add `self.request` (in the Backend)